src/server/vnparse.c

```diff
@@ -3426,15 +3961,6 @@ pbs_release_nodes_given_select(relnodes_input_t *r_input, relnodes_input_select_
 					}
 				}
 
-				if (parent_mom == NULL) { /* should not happen */
-
-					if ((err_msg != NULL) && (err_msg_sz > 0)) {
-						snprintf(err_msg, err_msg_sz, "no parent_mom for for %s", noden);
-        					log_event(PBSEVENT_DEBUG2, PBS_EVENTCLASS_JOB, LOG_DEBUG, r_input->jobid, err_msg);
-					}
-					goto release_nodes_exit;
-				}
-
```

Comment:
Even though this should not happen, why are we removing the check?

Reply:
Not removed. It just moved down to line #3996